pkgs/development/compilers/julia/1.6.nix

```diff
@@ -0,0 +1,164 @@
+{ lib, stdenv, fetchzip
+# build tools
+, gfortran, m4, makeWrapper, patchelf, perl, which, python3, cmake
+# libjulia dependencies
+, libunwind, readline, utf8proc, zlib, gmp
+# standard library dependencies
+, curl, fftwSinglePrec, fftw, libgit2, libnghttp2, mpfr, openlibm, openspecfun, pcre2
+# linear algebra
+, blas, lapack, arpack
+# Darwin frameworks
+, CoreServices, ApplicationServices
+}:
+
+assert (!blas.isILP64) && (!lapack.isILP64);
+
+with lib;
+
+let
+  majorVersion = "1";
+  minorVersion = "6";
+  maintenanceVersion = "0";
+  src_sha256 = "14qhd0vp2y9c6126niz37vkz3n7skd6h565fai3khhd0li63vka6";
+  version = "${majorVersion}.${minorVersion}.${maintenanceVersion}";
+in
+
+stdenv.mkDerivation rec {
+  pname = "julia";
+  inherit version;
+
+  src = fetchzip {
+    url = "https://github.com/JuliaLang/julia/releases/download/v${version}/julia-${version}-full.tar.gz";
+    sha256 = src_sha256;
+  };
+
+  patches = [
+    ./use-system-utf8proc-julia-1.3.patch
+
+    # Julia recompiles a precompiled file if the mtime stored *in* the
+    # .ji file differs from the mtime of the .ji file.  This
+    # doesn't work in Nix because Nix changes the mtime of files in
+    # the Nix store to 1. So patch Julia to accept mtimes of 1.
+    ./allow_nix_mtime.patch
+
+    # There's a weird error where n_succeeded would fail here: https://github.com/JuliaLang/julia/blame/3276c11b7e6ebf0c3867022765c8b233c0cbefe5/contrib/generate_precompile.jl#L378
+    # but this has since been fixed upstream. This patch is the output of
+    #   $ git format-patch -1 93b89b9486c2f5b6f780d59405b26e810412308f
+    # but with some healthy redaction to help the patch application work.
+    ./0001-reduce-precompile-failure-severity-to-a-warning-3990.patch
+  ];
+
+  postPatch = ''
+    patchShebangs . contrib
+  '';
+
+  dontUseCmakeConfigure = true;
+  enableParallelBuilding = true;
+
+  buildInputs = [ blas lapack libunwind utf8proc ]
+    ++ optionals stdenv.isDarwin [ CoreServices ApplicationServices ];
+
+  nativeBuildInputs = [ cmake curl gfortran m4 makeWrapper patchelf perl python3 which ];
+
+  makeFlags =
+    let
+      arch = head (splitString "-" stdenv.system);
+      march = {
+        x86_64 = stdenv.hostPlatform.gcc.arch or "x86-64";
+        i686 = "pentium4";
+        aarch64 = "armv8-a";
+      }.${arch}
+              or (throw "unsupported architecture: ${arch}");
+      # Julia requires Pentium 4 (SSE2) or better
+      cpuTarget = { x86_64 = "x86-64"; i686 = "pentium4"; aarch64 = "generic"; }.${arch}
+                  or (throw "unsupported architecture: ${arch}");
+    # Julia applies a lot of patches to its dependencies, so for now do not use the system LLVM
+    # https://github.com/JuliaLang/julia/tree/master/deps/patches
+    in [
+      "ARCH=${arch}"
+      "MARCH=${march}"
+      "JULIA_CPU_TARGET=${cpuTarget}"
+      "PREFIX=$(out)"
+      "prefix=$(out)"
+      "SHELL=${stdenv.shell}"
+
+      "USE_SYSTEM_BLAS=1"
```

Comment:
Any Darwin users willing to check this? I think this was made conditional on «not Darwin» for 1.5

Reply:
Looks like that change was made after I started work on this PR: https://github.com/NixOS/nixpkgs/commit/6a2a2d6503f450941bd471d9933e55c35e80625e. Should I make that same edit for `1.6.nix`?

Reply:
I don't really think there is a difference here between Julia versions, so probably it's better to apply this change here as well.

Reply:
Ok, done in 155f9f8!